dartagnan/src/main/java/com/dat3m/dartagnan/wmm/axiom/Empty.java

```diff
@@ -13,20 +11,20 @@ public Empty(Relation rel) {
         super(rel);
     }
 
-    @Override
-    public TupleSet getEncodeTupleSet(){
-        return rel.getMaxTupleSet();
-    }
+	@Override
+	public TupleSet getEncodeTupleSet(){
+		return new TupleSet();
+	}
 
-    @Override
-    public BooleanFormula consistent(SolverContext ctx) {
-    	BooleanFormulaManager bmgr = ctx.getFormulaManager().getBooleanFormulaManager();
-		BooleanFormula enc = bmgr.makeTrue();
-        for(Tuple tuple : rel.getEncodeTupleSet()){
-            enc = bmgr.and(enc, bmgr.not(rel.getSMTVar(tuple, ctx)));
-        }
-        return enc;
-    }
+	@Override
+	public TupleSet getDisabledSet() {
+		return new TupleSet(rel.getMaxTupleSet());
```

Comment:
It is, as I wrote `Relation#disable(TupleSet)` to expect ownership on its parameter.  You cold say it is passed by value, instead of reference.  Alternatively, all other intermediate results would need to be copied every time.